compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3412,65 +3403,61 @@ object Parsers {
-          val parents =
-            if in.token == COLON then
-              in.nextToken()
-              if in.token == LBRACKET
-                 || in.token == LPAREN && followingIsParamOrGivenType()
-              then
-                parseParams(isExtension = true)
-                Nil
-              else
-                constrApps(commaOK = true, templateCanFollow = true)
-            else if in.token == SUBTYPE then
-              if !mods.is(Inline) then
-                syntaxError("`<:' is only allowed for given with `inline' modifier")
-              in.nextToken()
-              TypeBoundsTree(EmptyTree, toplevelTyp()) :: Nil
-            else if name.isEmpty && !hasExtensionParams then
-              constrApps(commaOK = true, templateCanFollow = true)
+          val hasLabel = !name.isEmpty && in.token == COLON
+          if hasLabel then in.nextToken()
+          val tparams = typeParamClauseOpt(ParamOwner.Def)
+          val paramsStart = in.offset
+          val vparamss =
+            if in.token == LPAREN && followingIsParamOrGivenType()
+            then paramClauses()
             else Nil
-
-          if in.token == EQUALS && parents.length == 1 && parents.head.isType then
+          val isExtension = isIdent(nme.extended)
+          def checkAllGivens(vparamss: List[List[ValDef]], what: String) =
+            vparamss.foreach(_.foreach(vparam =>
+              if !vparam.mods.is(Given) then syntaxError(em"$what must be `given`", vparam.span)))
+          if isExtension then
+            if !name.isEmpty && !hasLabel then
+              syntaxError(em"name $name of extension clause must be followed by `:`", nameStart)
+            vparamss match
+              case (vparam :: Nil) :: vparamss1 if !vparam.mods.is(Given) =>
+                checkAllGivens(vparamss1, "follow-on parameter in extension clause")
+              case _ =>
+                syntaxError("extension clause must start with a single regular parameter", paramsStart)
             in.nextToken()
-            mods1 |= Final
-            DefDef(name, tparams, vparamss, parents.head, subExpr())
+            accept(WITH)
+            val (self, stats) = templateBody()
+            stats.foreach(checkExtensionMethod(tparams, _))
+            ModuleDef(name, Template(makeConstructor(tparams, vparamss), Nil, Nil, self, stats))
           else
-            parents match
-              case TypeBoundsTree(_, _) :: _ => syntaxError("`=' expected")
-              case _ =>
-            possibleTemplateStart()
-            if hasExtensionParams then
-              in.observeIndented()
+            checkAllGivens(vparamss, "parameter of given instance")
+            val parents =
+              if hasLabel then
+                constrApps(commaOK = true, templateCanFollow = true)
+              else if in.token == SUBTYPE then
+                if !mods.is(Inline) then
+                  syntaxError("`<:' is only allowed for given with `inline' modifier")
```

Comment:
Use `` `<:` `` instead of `` `<:' `` (same with inline) ? Both conventions seem to be used in the parser, but markdown-like syntax is probably more user-friendly.

Reply:
Let's discuss which one to use and then standardize. I'll open an issue.